spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/outbound/ZeroMqMessageHandler.java

```diff
@@ -122,6 +151,26 @@ public ZeroMqMessageHandler(ZContext context, String connectUrl, SocketType sock
 		Assert.hasText(connectUrl, "'connectUrl' must not be empty");
 	}
 
+	/**
+	 * Create an instance based on the provided {@link ZContext}, binding port and {@link SocketType}.
+	 * @param context the {@link ZContext} to use for creating sockets.
+	 * @param port the port to bind ZeroMq socket to over TCP.
+	 * @param socketType the {@link SocketType} to use;
+	 *    only {@link SocketType#PAIR}, {@link SocketType#PUB} and {@link SocketType#PUSH} are supported.
+	 * @since 6.2.6
+	 */
+	public ZeroMqMessageHandler(ZContext context, int port, SocketType socketType) {
+		this(context, socketType);
+		this.bindPort.set(port);
+		this.socketMono =
+				Mono.just(this.context.createSocket(this.socketType))
+						.publishOn(this.publisherScheduler)
+						.doOnNext((socket) -> this.socketConfigurer.accept(socket))
+						.doOnNext((socket) -> this.bindPort.set(bindSocket(socket, this.bindPort.get())))
```

Comment:
Well, actually we do this in the `start()`: 
```
this.socketMonoSubscriber = this.socketMono.subscribe();
```
So, all good, however looking into this one more time I believe that we have to always renew this `socketMono` in the `start()`. After the `stop()` the environment might be changed, so or now way to connect or bind.
Therefore every time when we start this component we have to renew socket.


I'm also thinking about DRY principle here.
Looks like we now have two socket `Mono` which has difference only in these lines:
```
.doOnNext((socket) -> socket.connect(connectUrl.get()))
```
in opposed to:
```
.doOnNext((socket) -> this.bindPort.set(bindSocket(socket, this.bindPort.get())))
```
Essentially `if..else` according to the server/client preferences.
This could be just extracted into a `private` method where we would do `connect` or `bind`. 
And `monoSocket` would be just a single instance in the `start` without any duplication.

WDYT?

Reply:
Completely agree with you. I was trying to touch less code as possible, but as you said this is the best approach